### PR TITLE
Support classes in `java.time` for `@Param` annotation

### PR DESCRIPTION
**Pull request by ghkim3221**
### Motivation

#2792

### Modification

- Add classes in `java.time` package to supported element types
  - `Duration`
  - `Instant`
  - `LocalDate`
  - `LocalDateTime`
  - `LocalTime`
  - `OffsetDateTime`
  - `OffsetTime`
  - `ZonedDateTime`
  - `ZoneId`
  - `ZoneOffset`
- `DocService` rendering:
  image

### CODE REVIEW

**Comment by ghkim3221** [comment]
`DocService` rendering:
image

File: core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolverTest.java [hunk 1]
```diff
@@ -724,10 +886,60 @@ public Cookies cookies() {
             return cookies;
         }
 
+        @Override
+        public Duration duration() {
+            return null;
```
Comment (ikhoon) [inline core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolverTest.java:891]: Question: Why these getters return null?

File: site/src/pages/docs/server-annotated-service.mdx [hunk 2]
```diff
@@ -187,7 +187,17 @@ one of the following supported types:
...
+- `Instant`
+- `LocalDate`
+- `LocalDateTime`
+- `LocalTime`
+- `OffsetDateTime`
+- `OffsetTime`
 - `Period`
+- `ZonedDateTime`
+- `ZoneId`
+- `ZoneOffset`
```
Comment (trustin) [inline site/src/pages/docs/server-annotated-service.mdx:200]: Could we make this list compact? e.g.

```
- Primitive types and their wrapper types: `boolean`, `byte`, .... and `double`
- Well-known JDK classes: `String`, `Enum`, `UUID`, ... and `ZoneOffset`
```

File: core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolverTest.java (hunk 1 above)
Comment (ghkim3221) [reply core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolverTest.java:891]: My mistake :( I will fix soon.

**Comment by trustin** [comment]
Sorry about the conflict, @ghkim3221. Would you mind merging master?

File: site/src/pages/docs/server-annotated-service.mdx (hunk 2 above)
Comment (ghkim3221) [reply site/src/pages/docs/server-annotated-service.mdx:200]: How about this? `Well-known JDK classes: String, Enum, UUID, ... and ZoneOffset` can be confused.

e.g:
```
- Primitive types and their wrapper types:
  - `boolean` or `Boolean`
  - `byte` or `Byte`
  - `short` or `Short`
  - `integer` or `Integer`
  - `long` or `Long`
  - `float` or `Float`
  - `double` or `Double`
- Well-known JDK classes:
  - `String`
  - `Enum`
  - `UUID`
  - `java.time` package classes
```

But, this PR does not support some `java.time` classes (`Clock`, `MonthDay`, `Year`). In this aspect, it seems better to show clearly.
```
  - `java.time` package classes
    -  `Instant`
    - `Duration` and  `Period`
    - `LocalDate`, `LocalDateTime` and `LocalTime`, 
    - `OffsetDateTime` and `OffsetTime`
    - `ZonedDateTime`, `ZoneId` and `ZoneOffset`
```

How do you think?

**Comment by ghkim3221** [comment]
> Sorry about the conflict, @ghkim3221. Would you mind merging master?

Of course, I will rebase after discussion ends :)

File: site/src/pages/docs/server-annotated-service.mdx (hunk 2 above)
Comment (trustin) [reply site/src/pages/docs/server-annotated-service.mdx:200]: The nested list with detailed list of java.time classes sounds good to me.

**Comment by ghkim3221** [comment]
@trustin apply comments and rebase to `master` 🙂

**Comment by ghkim3221** [comment]
Oh, thanks 😅

**Comment by trustin** [comment]
Thanks a lot, @ghkim3221 ! :tada:

**Pull request merged**